docs/update_switcher.py

```diff
@@ -21,31 +22,32 @@ def main():
     with open(switcher_path) as f:
         switcher = json.load(f)
 
-    # Find index of 'latest' entry
-    latest_index = None
+    # first we get the version number of the previous version
     for i, version in enumerate(switcher):
         if version["version"] == "latest":
             latest_index = i
+            previous_version = re.search(r"latest \(([v.\d]+)\)", version["name"]).group(1)
+            if previous_version == args.version:
+                print(f"Version {args.version} already is the latest version. Exiting.")
+                return
+
+            # now replace the name of this one with the new version
+            switcher[i]["name"] = f"latest ({args.version})"
             break
-    if latest_index is None:
+    else:
```

Comment:
I didn't even realize this kind of `for` / `if` / `break` / `else` combo was possible. TIL, interesting. :+1: